### PR TITLE
Filterfruit

### PR DESCRIPTION
**Pull request by carodriguezp**
please review

### CODE REVIEW

File: src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import FruitListContainer from "../FruitListContainer";
 
 jest.mock("../../../../../services/api"); //route to be mocked
 
+////////////FROM FRUIT LIST COMPONENT ////////////////
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: Missing tests?

File: src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx [hunk 2]
```diff
@@ -1,19 +1,31 @@
-import React from "react";
+import FormControl from "react-bootstrap/FormControl";
+import FormLabel from "react-bootstrap/FormLabel";
+import { StyledForm, StyledFormGroup } from "../../../../../../App.style";
+import { useEffect, useState } from "react";
+
+function Filter(props: { handleFilterFruit: (x: string) => void }) {
+  const { handleFilterFruit } = props;
+
+  const [value, setValue] = useState(""); //es filterfruits
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx]: Remove comment

File: src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+      <StyledFormGroup>
+        <FormLabel>Name</FormLabel>{" "}
+        <FormControl
+          data-testid="input"
+          type="text"
+          placeholder="Strawberry"
+          value={value}
+          onChange={(ev) => setValue(ev.target.value)}
+        />
+        <FormControl.Feedback />
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx]: Is the feedback subcomponent needed?

File: src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+//CASOS DEL TEST
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import Filter from "../Filter";
+import userEvent from "@testing-library/user-event";
+
+//DEL FILTER
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx]: Remove. Comments

File: src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+//PONERLE TEST ID AL FORMCONTROL
+
+describe("Filter", () => {
+  const filterProps = {
+    handleFilterFruit: jest.fn(),
+  };
+
+  const userValue = "Banana";
+
+  describe("When user writes in imput", () => {
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx]: Typo `input`*

File: src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    it("calls handleFilterFruit function to set value and set imput value", () => {
+      render(<Filter {...filterProps} />);
+
+      const input = screen.getByTestId("input");
+      userEvent.type(input, userValue); //.type simula que el usuario teclea--necesita 2 argumentos(un elemento y un text)
+
+      expect(input).toHaveValue(userValue);
+      expect(filterProps.handleFilterFruit).toHaveBeenCalledWith(userValue);
+
+      //screen.debug();
```
Comment (davepadrino) [inline src/components/main/fruits/fruitsListSection/components/filters/__test__/Filter.test.tsx]: Remove comment

File: src/setupTests.ts [hunk 7]
```diff
@@ -3,3 +3,4 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom';
+import "@testing-library/jest-dom/extend-expect";
```
Comment (davepadrino) [inline src/setupTests.ts:6]: Is this needed?

**Review by davepadrino — COMMENTED** [review]
Draft PR, will approve once ready to review

File: src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx (hunk 3 above)
Comment (carodriguezp) [reply src/components/main/fruits/fruitsListSection/components/filters/Filter.tsx]: No, due to the last decisions made.
Deleted

File: src/setupTests.ts (hunk 7 above)
Comment (carodriguezp) [reply src/setupTests.ts:6]: Yes

**Comment by carodriguezp** [comment]
Please @davepadrino check last updates in code.

Tests passed and structure was updated.

Bear in mind that commented lines of code will be deleted (they are still there to check faster in case of more modifications are needed).

If all is ok, could we proceed with pull request and merge?

File: src/components/main/fruits/fruitsListSection/FruitListContainer.tsx [hunk 8]
```diff
@@ -17,19 +19,22 @@ function FruitListContainer() {
...
-      return fruit.name.toLowerCase().includes(filterFruit.toLowerCase());
-    });
-
   useEffect(() => {
     fetchFruits();
   }, []);
 
+  useEffect(() => {
+    if (fruitsList) {
+      const x = fruitsList.filter((fruit) => {
```
Comment (davepadrinoxing) [inline src/components/main/fruits/fruitsListSection/FruitListContainer.tsx]: what is `x`? use meaningful names

File: src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx [hunk 9]
```diff
@@ -1,85 +1,92 @@
...
 import FruitListContainer from "../FruitListContainer";
+import userEvent from "@testing-library/user-event";
 import { HashRouter } from "react-router-dom";
-import FruitList from "../FruitList";
+import { act } from "react";
+import { toBeInTheDocument } from "@testing-library/jest-dom/types/matchers";
 
 jest.mock("../../../../../services/api"); //route to be mocked
 
+//describe("", () => {})
```
Comment (davepadrinoxing) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: is this needed?

File: src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx [hunk 10]
```diff
@@ -1,137 +1,244 @@
...
+    });
 
-        it("displays an error", async () => {
+    describe("When input from user is received", () => {
+      describe("When succesfull finding data", () => {
+        // beforeEach(() => {
+        //   jest
+        //     .spyOn(myService, "getDataFromApi")
+        //     .mockResolvedValue(mockFruitListFromApi);
+        // });
```
Comment (davepadrinoxing) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: don't think we need this anymore, do we?

File: src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx [hunk 11]
```diff
@@ -1,137 +1,244 @@
...
           // eslint-disable-next-line testing-library/no-unnecessary-act
           act(() => {
             userEvent.type(input, userSearch); //.type simulates user typing--needs 2 arguments (an element and a text)
           });
           await waitFor(() => {
-            foundFruit = screen.queryByTestId(userSearch);
+            foundFruit = screen.getByTestId(userSearch);
           });
 
           //screen.debug();
```
Comment (davepadrinoxing) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: remove commented line

Comment (davepadrinoxing) [reply src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: here and everywhere in this file

File: src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx [hunk 12]
```diff
@@ -1,137 +1,244 @@
...
+});
 
-  describe("when not succesfull", () => {
-    beforeEach(() => {
-      jest.spyOn(myService, "getDataFromApi").mockResolvedValue([]);
-    });
-
-    it("displays an error", async () => {
-      render(<FruitListContainer />);
+////////////////////////////////////////////////////
```
Comment (davepadrinoxing) [inline src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx]: i know you mentioned something regarding the commented code, but do we really needed to merge it? don't think so. Let's get rid of it :)

**Review by davepadrinoxing — APPROVED** [review]
GREAT JOB 🎖️ 

Let's address the comments and we can move on

**Pull request merged**